subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ApplicationPluginUnixShellsIntegrationTest.groovy

```diff
@@ -32,20 +33,7 @@ class ApplicationPluginUnixShellsIntegrationTest extends AbstractIntegrationSpec
         }
     }
 
-    static boolean shellAvailable(String shellCommand) {
-        return TestPrecondition.UNIX_DERIVATIVE.isFulfilled() && locate(shellCommand)
-    }
-
-    static File locate(String shellCommand) {
-        return [
-                new File("/bin/$shellCommand"),
-                new File("/usr/bin/$shellCommand"),
-                new File("/usr/local/bin/$shellCommand"),
-                new File("/opt/local/bin/$shellCommand")
-        ].find { it.exists() }
-    }
-
-    @Requires(adhoc = { ApplicationPluginUnixShellsIntegrationTest.shellAvailable("bash") })
+    @Requires([UnitTestPreconditions.UnixDerivative, PluginTestPreconditions.BashAvailable])
```

Comment:
By the current implementation, that could be the case:

https://github.com/gradle/gradle/blob/aa94720639e65a390ad6f45eaf05a380e4db1513/subprojects/plugins/src/testFixtures/groovy/org/gradle/test/preconditions/PluginTestPreconditions.groovy#L22-L29

Why? Because even if you have MSYS2, it's going to be on a path that is not going to be detected by our current logic.

Reply:
Attempted it in 9dd64220

Reply:
Seems to work, the [PRF](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/65572312?hideProblemsFromDependencies=false&expandBuildTestsSection=false&hideTestsFromDependencies=false&expandBuildChangesSection=false) is green.